### PR TITLE
A preliminary support for RiSAWOZ dataset

### PR DESCRIPTION
**Pull request by Magnetic2014**

### CODE REVIEW

File: dialogues/risawoz/src/convert.py [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+            turn_api_text = []
+            for d, sv in dialog_state.items():
+                if d in final_dialog_state.keys():
+                    final_dialog_state[d].update(sv)
+                else:
+                    final_dialog_state[d] = sv
+                # build turn API text
+                try:
+                    if len(turn["db_results"]) > 1:
+                        first_result = eval(turn["db_results"][1].replace("true", "True").replace("false", "False"))
```
Comment (Mehrad0711) [inline dialogues/risawoz/src/convert.py:207]: In bitod, the agent only needs the first result to continue the conversation. In other words, they rank the results based on user rating and return the first. This way you always get the same result (output) for the same constraints (input). Is that also the case in Risawoz? Does agent always look at the first result to formulate its response?

File: dialogues/risawoz/src/knowledgebase/en_zh_mapping.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
+class RisaWOZMapping(object):
+    def __init__(self):
+        # currently untranslated
+        # TODO: this mapping should only include slots that are required to make an api call for the domain. It should not include all the slots.
```
Comment (Mehrad0711) [inline dialogues/risawoz/src/knowledgebase/en_zh_mapping.py:4]: This is an important TODO. In BiToD, they have this concept of `required_slots` which is defined as slots needed to be filled out before agent can make an api call. These are a subset of all slots available to a domain. Can we define that mapping for Risawoz?

File: dialogues/risawoz/db/zh/attraction_db.json [hunk 3]
```diff
@@ -0,0 +1,562 @@
+[
+  {
```
Comment (Mehrad0711) [inline dialogues/risawoz/db/zh/attraction_db.json:2]: We need the database for English too

File: dialogues/bitod/src/knowledgebase/en_zh_mappings.py [hunk 4]
```diff
@@ -25,6 +26,13 @@ def read_require_slots():
     return require_slots
 
 
+risawoz_mapping = RisaWOZMapping()
```
Comment (Mehrad0711) [inline dialogues/bitod/src/knowledgebase/en_zh_mappings.py:29]: We need  the mapping for English too

**Comment by Mehrad0711** [comment]
I'm gonna merge this branch for now before it gets bigger! But the TODOs need to be addressed later.

File: dialogues/risawoz/src/knowledgebase/en_zh_mapping.py (hunk 2 above)
Comment (Magnetic2014) [reply dialogues/risawoz/src/knowledgebase/en_zh_mapping.py]: The required slots can be found in the original RiSAWOZ paper. I use basically the same translation in the paper for slots so they can be directly used.

File: dialogues/risawoz/db/zh/attraction_db.json (hunk 3 above)
Comment (Magnetic2014) [reply dialogues/risawoz/db/zh/attraction_db.json]: I just sent you an English version of database via email. However, there are some untranslated non-slot attributes and values, which may need some time to translate manually. You can have a try to see if the current version is good for training.

File: dialogues/risawoz/src/convert.py (hunk 1 above)
Comment (Magnetic2014) [reply dialogues/risawoz/src/convert.py:207]: Yes, the first result is always used in the conversation.

File: dialogues/bitod/src/knowledgebase/en_zh_mappings.py (hunk 4 above)
Comment (Magnetic2014) [reply dialogues/bitod/src/knowledgebase/en_zh_mappings.py:29]: I've created a PR to revise your machine-translated mapping. Please let me know if you need other alignment information.

**Pull request merged**